server/sonar-server/src/main/java/org/sonar/server/qualitygate/changeevent/QGChangeEventListener.java

```diff
@@ -20,7 +20,40 @@
 package org.sonar.server.qualitygate.changeevent;
 
 import java.util.Collection;
+import java.util.Collections;
+import java.util.Set;
+import org.sonar.api.rules.RuleType;
 
 public interface QGChangeEventListener {
-  void onChanges(Trigger trigger, Collection<QGChangeEvent> changeEvents);
+  /**
+   * @deprecated use {{@link #onIssueChanges(QGChangeEvent, Set)}} instead
+   */
+  // TODO remove this method and turn default method #onIssueChanges(QGChangeEvent, Set) into an interface method when
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Complete the task associated to this TODO comment. ([squid:S1135](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1135))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fsl%2FSONAR-10085%2Fupgrade_qg_change_event_api_with_issue_details&issues=AWCXVfHxqDrCo5ysfvWU&open=AWCXVfHxqDrCo5ysfvWU)

[ISSUE_KEY]: AWCXVfHxqDrCo5ysfvWU